src/main/scala/viper/silver/plugin/standard/inline/InlineErrorChecker.scala

```diff
@@ -81,13 +81,17 @@ trait InlineErrorChecker {
     */
   private[this] def isRecursivePred(predId: String, maybePredBody: Option[Node]): Boolean =
     maybePredBody.fold(false) { predBody =>
+      val hasDirectRecursiveCall = predBody match {
+        case PredicateAccessPredicate(PredicateAccess(_, name), _) => name == predId
+        case _ => false
+      }
       val subNodes = predBody.subnodes
       val existsAtTopLevelNode = subNodes.exists {
         case PredicateAccessPredicate(PredicateAccess(_, name), _) => name == predId
         case _ => false
       }
       lazy val isInChildNodes = subNodes.exists(child => isRecursivePred(predId, Some(child)))
```

Comment:
This could probably then be smth like `predBody.subnodes.exists(isRecursivePred(predId, Some(_)))`